php/class-ee-db.php

```diff
@@ -2,325 +2,482 @@
+		$this->offset = ' OFFSET ' . (string) $offset;
 
-		return false;
+		return $this;
 	}
 
 	/**
-	 * Delete data from table.
+	 * Insert row in table.
 	 *
-	 * @param        $where
+	 * @param array $data in key value pair.
+	 *
+	 * @throws Exception If no table or more than one tables are specified
 	 *
 	 * @return bool
 	 */
-	public static function delete( $where, $table_name = 'sites' ) {
+	public function insert( $data ) {
 
-		$conditions = array();
-		foreach ( $where as $key => $value ) {
-			$conditions[] = "`$key`='" . $value . "'";
+		$fields = implode( ', ', array_keys( $data ) );
+		$values = implode( ', ', array_fill( 0, count( $data ), '?' ) );
+
+		if ( empty( $this->tables ) ) {
+			throw new Exception( 'Insert: No table specified' );
 		}
 
-		$conditions   = implode( ' AND ', $conditions );
-		$delete_query = "DELETE FROM `$table_name` WHERE $conditions";
+		if ( strpos( $this->tables, ',' ) !== false ) {
+			throw new Exception( 'Insert: Multiple table specified' );
+		}
 
-		$delete_query_exec = self::$db->exec( $delete_query );
+		$query = "INSERT INTO $this->tables ($fields) VALUES ($values);";
 
-		if ( ! $delete_query_exec ) {
-			EE::debug( self::$db->lastErrorMsg() );
-		} else {
-			return true;
+		$pdo_statement = self::$pdo->prepare( $query );
+		$bindings      = array_values( $data );
+
+		foreach ( $bindings as $key => $value ) {
+			$pdo_statement->bindValue( $key + 1, $value );
 		}
 
-		return false;
+		$result = $pdo_statement->execute();
+
+		if ( ! $result ) {
+			EE::debug( self::$pdo->errorInfo() );
+
+			return false;
+		}
+
+		return self::$pdo->lastInsertId();
 	}
 
 	/**
-	 * Check if a site entry exists in the database.
+	 * Update row in table.
 	 *
-	 * @param String $site_name Name of the site to be checked.
+	 * @param array $values Associative array of columns and their values
 	 *
-	 * @return bool Success.
+	 * @throws Exception If no table are specified or multiple table are specified or no where clause are specified
+	 *
+	 * @return bool
 	 */
-	public static function site_in_db( $site_name ) {
+	public function update( $values ) {
+		if ( empty( $this->tables ) ) {
+			throw new Exception( 'Update: No table specified' );
+		}
 
-		if ( empty ( self::$db ) ) {
-			self::init_db();
+		if ( empty( $this->where ) ) {
+			throw new Exception( 'Update: No where clause specified' );
 		}
 
-		$site = self::select( array( 'id' ), array( 'sitename' => $site_name ) );
+		if ( strpos( $this->tables, ',' ) !== false ) {
+			throw new Exception( 'Update: Multiple table specified' );
+		}
 
-		if ( $site ) {
-			return true;
-		} else {
+		if ( empty( $values ) ) {
 			return false;
 		}
+
+		$set_keys     = array_keys( $values );
+		$set_bindings   = array_values( $values );
+		$where_bindings = $this->where['bindings'];
+
+		$set_clause = implode( $set_keys, ' = ?, ' ) . ' = ?';
+
+		$query         = "UPDATE $this->tables SET $set_clause{$this->where['query_string']}";
```

Comment:
I think here you should also add the modified_time for site table. or you're doing it in all places where you update any data in site table?

Reply:
`class-ee-db.php` is a generic class and will be used by all tables. So instead of adding modified_date here, I will add it in `Site` model's save() method. (That's what we're using everywhere to update a site)

Reply:
Yes, that should be the case.